apps/sms/views/shared/js/activity_handler.js

```diff
@@ -356,20 +356,22 @@ var ActivityHandler = {
             data: { id, threadId }
           };
 
-          return titlePromise.then((title) => {
-            var notification = new Notification(title, options);
-            notification.addEventListener('click', goToMessage);
-            releaseWakeLock();
-
-            // Close notification if we are already in thread view and view
-            // become visible.
-            if (document.hidden && threadId === Threads.currentId) {
-              document.addEventListener('visibilitychange',
-                function onVisible() {
-                  document.removeEventListener('visibilitychange', onVisible);
-                  notification.close();
-              });
-            }
+          titlePromise.then(function(title) {
+            NotificationHelper.send(title, options).then(
```

Comment:
Because we don't need the title inside the notification's promise chain, I'd do something with less indentation like this:

```javascript
return titlePromise.then(
  (title) => NotificationHelper.send(title, options)
).then((notification) => {
  notification.addEventListener(...);
  ...
});
```

(we return because it's good practice ... and could make unit tests easier in the future when we'll return promises from all asynchronous functions, which is not the case yet here)